src/SplitClient.tsx

```diff
@@ -23,8 +23,13 @@ export class SplitComponent extends React.Component<IUpdateProps & { factory: Sp
   // Using `getDerivedStateFromProps` since the state depends on the status of the client in props, which might change over time.
   // It could be avoided by removing the client and its status from the component state.
   // But it implies to have another instance property to use instead of the state, because we need a unique reference value for SplitContext.Producer
-  static getDerivedStateFromProps(props: ISplitClientProps & { factory: SplitIO.ISDK | null, client: SplitIO.IClient | null }, state: ISplitContextValues) {
-    const { client, factory } = props;
+  static getDerivedStateFromProps(props: ISplitClientProps & { factory: SplitIO.IBrowserSDK | null, client: SplitIO.IBrowserClient | null }, state: ISplitContextValues) {
+    const { client, factory, attributes } = props;
+    // @TODO check if apply over client or over stage.client
+    if (attributes)
```

Comment:
Due to the declarative/functional nature of React, maybe we should consider updating the logic into:
```
client?.clearAttributes();
if (attributes) client?.setAttributes(attributes);
```

So that if the `attributes` props change, old attributes that are not included in the new attributes object, are removed from the client. 
Same should apply for `useClient`.

Also, remove the `@TODO` comment.

Reply:
Also, it would be great to validate this fix with a test that fails without the fix.